platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/ReconnectBench.java

```diff
@@ -71,62 +71,76 @@ public class ReconnectBench extends VirtualMapBaseBench {
     }
 
-    @Override
-    public void beforeTest(String name) {
-        super.beforeTest(name);
-        // Use a different MerkleDb instance for every test run. With a single instance,
-        // even if its folder is deleted before each run, there could be background
-        // threads (virtual pipeline thread, data source compaction thread, etc.) from
-        // the previous run that re-create the folder, and it results in a total mess
-        final Path merkleDbPath = getTestDir().resolve("merkledb" + dbIndex++);
-        MerkleDb.setDefaultPath(merkleDbPath);
-    }
-
     /**
      * Builds a VirtualMap populator that is able to add/update, as well as remove nodes (when the value is null.)
      * Note that it doesn't support explicitly adding null values under a key.
      *
-     * @param map a VirtualMap instance
+     * @param mapRef a reference to a VirtualMap instance
      * @return a populator for the map
      * @param <K> key type
      * @param <V> value type
      */
     private static <K extends VirtualKey, V extends VirtualValue> BiConsumer<K, V> buildVMPopulator(
-            final VirtualMap<K, V> map) {
+            final AtomicReference<VirtualMap<K, V>> mapRef) {
         return (k, v) -> {
             if (v == null) {
-                map.remove(k);
+                mapRef.get().remove(k);
             } else {
-                map.put(k, v);
+                mapRef.get().put(k, v);
             }
         };
     }
 
-    @Setup(Level.Invocation)
-    public void setupInvocation() {
+    /** Generate a state and save it to disk once for the entire benchmark. */
+    @Setup
+    public void setupBenchmark() {
         beforeTest("reconnect");
+        updateMerkleDbPath();
 
-        teacherMap = createEmptyMap("teacher");
-        learnerMap = createEmptyMap("learner");
+        final AtomicReference<VirtualMap<BenchmarkKey, BenchmarkValue>> teacherRef =
+                new AtomicReference<>(createEmptyMap("teacher"));
+        final AtomicReference<VirtualMap<BenchmarkKey, BenchmarkValue>> learnerRef =
+                new AtomicReference<>(createEmptyMap("learner"));
 
         final Random random = new Random(randomSeed);
         new StateBuilder<>(BenchmarkKey::new, BenchmarkValue::new)
                 .buildState(
                         random,
-                        numRecords,
+                        (long) numRecords * (long) numFiles,
                         teacherAddProbability,
                         teacherRemoveProbability,
                         teacherModifyProbability,
-                        buildVMPopulator(teacherMap),
-                        buildVMPopulator(learnerMap));
+                        buildVMPopulator(teacherRef),
+                        buildVMPopulator(learnerRef),
+                        i -> {
+                            if (i % numRecords == 0) {
+                                System.err.printf("Copying files for i = %,d\n", i);
```

Comment:
Debug output?

Reply:
I left it here intentionally. This output helps track the progress of building a state which is useful for larger states. E.g. the current 50M state takes around 10-15 minutes to build on my Mac, so it's nice to have a visual feedback when running this benchmark.